lib/core/src/Cardano/DB/Sqlite.hs

```diff
@@ -213,28 +211,105 @@ startSqliteBackend
     -> Migration
     -> Tracer IO DBLog
     -> Maybe FilePath
-    -> IO (Either MigrationError (Quantity "migration" Int, SqliteContext))
-startSqliteBackend manualMigration migrateAll trace fp = do
-    backend <- createSqliteBackend trace fp manualMigration (queryLogFunc trace)
+    -> IO (Either MigrationError SqliteContext)
+startSqliteBackend manualMigration autoMigration trace fp = do
+    (backend, connection) <-
+        createSqliteBackend trace fp manualMigration (queryLogFunc trace)
     lock <- newMVar ()
     let traceRun = traceWith trace . MsgRun
     let observe :: IO a -> IO a
         observe = bracket_ (traceRun False) (traceRun True)
     let runQuery :: SqlPersistT IO a -> IO a
         runQuery cmd = withMVar lock $ const $ observe $ runSqlConn cmd backend
-    migrations <- runQuery (runMigrationQuiet migrateAll)
-        & tryJust (matchMigrationError @PersistException)
-        & tryJust (matchMigrationError @SqliteException)
-        & fmap join
-        :: IO (Either MigrationError [Text])
-    traceWith trace $ MsgMigrations (fmap length migrations)
+    autoMigrationResult <-
+        withForeignKeysDisabled trace connection
+            $ runQuery (runMigrationQuiet autoMigration)
+            & tryJust (matchMigrationError @PersistException)
+            & tryJust (matchMigrationError @SqliteException)
+            & fmap join
+    traceWith trace $ MsgMigrations $ fmap length autoMigrationResult
     let ctx = SqliteContext backend runQuery fp trace
-    case migrations of
+    case autoMigrationResult of
         Left e -> do
             destroyDBLayer ctx
             pure $ Left e
-        Right ms -> do
-            pure $ Right (Quantity (length ms), ctx)
+        Right _ -> pure $ Right ctx
+
+-- | Run the given task in a context where foreign key constraints are
+--   /temporarily disabled/, before re-enabling them.
+--
```

Comment:
Note that what's interesting here is not really the "what", but the "why". The comment is relatively redundant with the function's name, though I see you explained more about the "why" down below in the "ForeignKeySetting" documentation. Fair enough :+1: